### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by carusogabriel**
Together with PHPStorm and https://github.com/kalessil/phpinspectionsea, here are some improvements that we can make to Composer :smile:

### CODE REVIEW

File: src/Composer/Installer.php [hunk 1]
```diff
@@ -1033,11 +1033,8 @@ private function processDevPackages($localRepo, $pool, $policy, $repositories, $
                         $package->setReplaces($newPackage->getReplaces());
                     }
 
-                    if ($task === 'force-updates' && $newPackage && (
-                        (($newPackage->getSourceReference() && $newPackage->getSourceReference() !== $package->getSourceReference())
-                            || ($newPackage->getDistReference() && $newPackage->getDistReference() !== $package->getDistReference())
-                        )
-                    )) {
+                    if ($task === 'force-updates' && $newPackage && ($newPackage->getSourceReference() && $newPackage->getSourceReference() !== $package->getSourceReference())
```
Comment (stof) [inline src/Composer/Installer.php]: Are you sure about this change ? I don't think it is right regarding the operator precedence

**Comment by carusogabriel** [comment]
@alcohol Done. I've also squashed the commits. Should be RTM, thank you :smile:

**Comment by alcohol** [comment]
Thanks

File: src/Composer/Installer.php (hunk 1 above)
Comment (stof) [reply src/Composer/Installer.php]: my concern is that without parenthesis, I suspect the initial conditions only apply to the left-side of the `||` (which your var-dumps don't test, as they make them always `true`)

Comment (Seldaek) [reply src/Composer/Installer.php]: Fixed in https://github.com/composer/composer/commit/d5a9d86ee4e934d06073521e630556d4f337629c for readability + clarity, no matter if it worked or not.

**Pull request merged**